### PR TITLE
Add support for files in generic parser

### PR DESCRIPTION
**Pull request by damiencarol**
Add support for a new attributes that accept files as Base64 strings.

Ex:
```JSON
{
    "title": "My wonderfull report",
    "findings": [
        {
            "title": "Vuln with image",
            "description": "Some very long description",
            "severity": "Medium",
            "files": [
                {
                    "title": "Screenshot from 2017-04-10 16-54-19.png",
                    "data": "iVBORw0KGgoAAAANSUhEUgAABWgAAAK0CAIAAAARSkPJAAAAA3N<...>TkSuQmCC"
                }
            ]
        }
    ]
}
```

image

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Any thoughts on what happens with lots of (big) files?

**Comment by damiencarol** [comment]
@valentijnscholten no idea. I only touched the generic parser. 🤔 
is it only in this PR?

**Comment by valentijnscholten** [comment]
I think this is the first parser doin it, or at least unsaved_files is new loading all files in memory. Not easy to avoid with our current parser contact, just wondering what would happen if somebody uploads 500M of screenshots.

**Comment by damiencarol** [comment]
oh. I see.
I thing today we limit to 3 files.
And most of the screenshots I saw used by InfoSec teams where PNG or BMP of few hundreds bytes.

**Comment by StefanFl** [comment]
Please adjust the example in https://defectdojo.github.io/django-DefectDojo/integrations/import/#generic-findings-import, so that we have a documentation.

Do we have an overall limit for the size of a file to upload?

**Comment by damiencarol** [comment]
mmhh I think it will be the limit of the size of the report. Django limit the upload anyway.
I don't think more checks are needed in the parser or (re)importer

**Comment by valentijnscholten** [comment]
Some docs would be nice, otherwise good to go.

File: docs/content/en/integrations/import.md [hunk 1]
```diff
@@ -461,6 +461,29 @@ Example of JSON format:
 }
 ```
 
+This parser support an attributes that accept files as Base64 strings. These files will be created as `UploadFile`
+
+Ex:
```
Comment (StefanFl) [inline docs/content/en/integrations/import.md:466]:
```suggestion
Example:
```

File: docs/content/en/integrations/import.md [hunk 2]
```diff
@@ -461,6 +461,29 @@ Example of JSON format:
...
+
+```JSON
+{
+    "title": "My wonderfull report",
```
Comment (StefanFl) [inline docs/content/en/integrations/import.md:470]:
```suggestion
    "title": "My wonderful report",
```

File: docs/content/en/integrations/import.md [hunk 3]
```diff
@@ -461,6 +461,29 @@ Example of JSON format:
 }
 ```
 
+This parser support an attributes that accept files as Base64 strings. These files will be created as `UploadFile`
```
Comment (StefanFl) [inline docs/content/en/integrations/import.md:464]: Where does the term `UploadFile` come from? In the UI they are referenced as "Files". 

Suggestion:

```suggestion
This parser support an attributes that accept files as Base64 strings. These files are attached to the respective findings.
```

**Review by StefanFl — CHANGES_REQUESTED** [review]

**Comment by mtesauro** [comment]
I know @Maffooch updated/re-vamped file uploads a while ago (maybe a year ago) - just wondering if this uses the same location & process to save the files.

One of the clean-up things Cody did was to unify how files were saved.  Prior to that, uploaded finding images were stored differently than threat model uploads so it would be good to avoid introducing a new file saving mechanism as part of this PR.

**Comment by damiencarol** [comment]
@mtesauro I used the new data structure

**Review by StefanFl — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]
Cool feature

**Pull request merged**